Improve documentation, add references to redis-macros

daniel7grant opened the pull request:
This is made in reference of [#740](https://github.com/redis-rs/redis-rs/issues/740#issuecomment-1409925147).

The crate [redis-macros](https://github.com/daniel7grant/redis-macros/) adds derive macros for `FromRedisValue` and `ToRedisArgs`, as well as a wrapper type `Json` to unwrap RedisJSON command results. I added suggestions for users to use these macros at several places in the documentation where types were mentioned. I fixed the documentation code blocks in `JsonCommands`, and added more information about RedisJSON results, and how to unwrap it with `Json`.

Feel free to edit if you find any inaccuracies.

djc commented:
I think linking to your crate should be restricted to the README and maybe one other high-level section in the documentation. Maybe we can then still have some of these other parts of the documentation link to that section?

daniel7grant commented:
I removed the references from the lower-level documentation. I left the fixes of the documentation I did for the RedisJSON section, it seemed to contain some outdated code snippets. I also added that the responses will be wrapped in a `Vec`, I think this is a useful information for people working with RedisJSON. I hope this is okay.

djc reviewed (approved):
Looks great, thanks! @jaymell any thoughts?

daniel7grant commented:
Sorry for not getting to this faster, but I fixed the test and lint errors. Is there anything else for me to do to move this PR along?

jaymell commented:
@daniel7grant I'd actually prefer it if we leave the doc changes out of `lib.rs`. Do you mind removing those, and we can merge the rest of the changes?

daniel7grant commented:
I'm somewhat concerned because this change eliminates the bulk of the documentation I added, except a line in the README. Nevertheless if you prefer it this way, then it's fine by me.

jaymell commented:
Thanks, I appreciate it. Your points are noted, I just wanted to keep a reference out of the main docs at this point, as I personally haven't vetted/looked at the crate.

daniel7grant commented:
I understand, my crate is far from being stable, I just hope it will help some people. Please tell me, if you can get to look over the crate and have any feedback. Thanks for the help!

thomaseizinger commented:
> @daniel7grant I'd actually prefer it if we leave the doc changes out of `lib.rs`. Do you mind removing those, and we can merge the rest of the changes?

Just wanted to chime in here and say that a pointer to `redis-macros` in lib.rs would have saved me a fair bit of time. At the moment, it is not very discoverable but it is very useful once you've found it!
